models/userStatus.js

```diff
@@ -325,6 +326,52 @@ const updateStatusOnTaskCompletion = async (userId) => {
   }
 };
 
+const getUsersWithoutAssignedOrInProgressTasks = async () => {
+  const usersWithoutAssignedOrInProgressTasks = [];
+  const usersNotProcessed = [];
+  let errorCount = 0;
+  let discordActiveNonArchivedUsersQuerySnapshot;
+  try {
+    discordActiveNonArchivedUsersQuerySnapshot = await usersCollection
+      .where("roles.in_discord", "==", true)
+      .where("roles.archived", "==", false)
+      .get();
```

Comment:
I think if `in_discord` is true, the `is_archived` is anyway going to be false?
Or when can be the case where in_discord is true and is_archived is also true?

Point is, major factor to be considered is if the user is in discord server or not. because end of the we are creating the tasks and everything for our discord users only, correct? what do you think?

Reply:
Ok, but something to think about going forward. Let's start this discussion in the discord channel

Reply:
can you approve please @sumitd94 if rest looks good.

Reply:
Since all the other comments are resolved and due to the urgency of this feature. I am resolving this and merging the PR.